internal/webhook/v1beta1/glance_webhook.go

```diff
@@ -0,0 +1,124 @@
+
+// Default implements webhook.CustomDefaulter so a webhook will be registered for the Kind Glance.
+func (d *GlanceCustomDefaulter) Default(_ context.Context, obj runtime.Object) error {
+	glance, ok := obj.(*glancev1beta1.Glance)
+
+	if !ok {
+		return fmt.Errorf("%w: got %T", errUnexpectedGlanceObjectType, obj)
+	}
+	glancelog.Info("Defaulting for Glance", "name", glance.GetName())
+
+	glance.Default()
+	return nil
+}
+
+// TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
+// NOTE: The 'path' attribute must follow a specific pattern and should not be modified directly here.
+// Modifying the path for an invalid path can cause API server errors; failing to locate the webhook.
+// +kubebuilder:webhook:path=/validate-glance-openstack-org-v1beta1-glance,mutating=false,failurePolicy=fail,sideEffects=None,groups=glance.openstack.org,resources=glances,verbs=create;update,versions=v1beta1,name=vglance-v1beta1.kb.io,admissionReviewVersions=v1
+
+// GlanceCustomValidator struct is responsible for validating the Glance resource
+// when it is created, updated, or deleted.
+//
+// NOTE: The +kubebuilder:object:generate=false marker prevents controller-gen from generating DeepCopy methods,
+// as this struct is used only for temporary operations and does not need to be deeply copied.
+type GlanceCustomValidator struct {
+	// TODO(user): Add more fields as needed for validation
+}
+
+var _ webhook.CustomValidator = &GlanceCustomValidator{}
+
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type Glance.
+func (v *GlanceCustomValidator) ValidateCreate(_ context.Context, obj runtime.Object) (admission.Warnings, error) {
+	glance, ok := obj.(*glancev1beta1.Glance)
+	if !ok {
+		return nil, fmt.Errorf("%w: got %T", errUnexpectedGlanceObjectType, obj)
+	}
+	glancelog.Info("Validation for Glance upon creation", "name", glance.GetName())
+
+	return glance.ValidateCreate()
+
+}
+
+// ValidateUpdate implements webhook.CustomValidator so a webhook will be registered for the type Glance.
+func (v *GlanceCustomValidator) ValidateUpdate(_ context.Context, oldObj, newObj runtime.Object) (admission.Warnings, error) {
+	glance, ok := newObj.(*glancev1beta1.Glance)
+	if !ok {
+		return nil, fmt.Errorf("%w: got %T", errUnexpectedGlanceObjectType, newObj)
+	}
+	glancelog.Info("Validation for Glance upon update", "name", glance.GetName())
+
+	return glance.ValidateUpdate(oldObj)
+
+}
+
+// ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type Glance.
+func (v *GlanceCustomValidator) ValidateDelete(_ context.Context, obj runtime.Object) (admission.Warnings, error) {
+	glance, ok := obj.(*glancev1beta1.Glance)
+	if !ok {
+		return nil, fmt.Errorf("%w: got %T", errUnexpectedGlanceObjectType, obj)
+	}
+	glancelog.Info("Validation for Glance upon deletion", "name", glance.GetName())
+
+	return nil, nil
```

Comment:
there is ValidateDelete() which is actually empty, but might be good to just call it in case its used in future?

Reply:
sure, I can add it.